### PR TITLE
Adding thailand AMLO designated list crawler

### PR DESCRIPTION
**Pull request by bgmello**
fixes opensanctions/crawler-planning#110

### CODE REVIEW

File: datasets/th/designated_person/th_designated_person.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  frequency: daily
+  start: 2024-04-08
+load_db_uri: ${OPENSANCTIONS_DATABASE_URI}
+summary: >
+  List of Designated Persons from Thailand's AMLO
+description: |
+  This dataset comprises information about designated persons published by
+  Thailand's Anti Money Laundring Office (AMLO)
+  under the Counter-Terrorism and Proliferation
+  Financing Act B.E. 2559 section 7
```
Comment (jbothma) [inline datasets/th/designated_person/th_designated_person.yaml:14]: Please wrap at 80 chars

```suggestion
  Financing Act B.E. 2559 section 7
  
  According to the act:
  
    > ...[R]eporting entities and persons in possession of the assets of the designated persons [will be informed to]  Freeze the assets of the designated persons or a person acting on behalf of, or at the direction of, or an undertaking owned or controlled by such persons, directly or indirectly
```

File: datasets/th/designated_person/crawler.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    en_name = info_dict.pop("Individual/Entity Name (English)").strip()
+    th_name = info_dict.pop("Individual/Entity Name (Thailand)").strip()
+
+    entity = context.make("Person")
+    entity.id = context.make_id(en_name)
+
+    entity.add("name", en_name, lang="en")
+    entity.add("name", th_name, lang="th")
+    entity.add("topics", "crime")
```
Comment (jbothma) [inline datasets/th/designated_person/crawler.py:29]: I think this is enough of a standard financial sanction to use the sanction topic.
```suggestion
    entity.add("topics", "sanction")
```

**Review by jbothma — CHANGES_REQUESTED** [review]
Very nice, thank you! A few small things. Also please take a look and address the lint issues identified in the lint workflow.

File: datasets/th/designated_person/crawler.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        entity.add("address", info_dict.pop("Address No.2"), lang="th")
+
+    if "Passport Number" in info_dict and info_dict["Passport Number"] != "":
+        entity.add("passportNumber", info_dict.pop("Passport Number"))
+
+    sanction = h.make_sanction(
+        context, entity, key=info_dict.pop("Notification Number")
+    )
+
+    sanction.add("sourceUrl", url)
```
[the remark this answers is not in the document]
Comment (bgmello) [reply datasets/th/designated_person/crawler.py]: I also added the Notification Number to this.

[the remark this answers is not in the document]
Comment (jbothma) [reply datasets/th/designated_person/crawler.py]: I think that's ok - let's go with this for now.

**Pull request merged**